extensions/machinepools/machinepools.go

```diff
@@ -54,6 +56,10 @@ func MatchNodeRolesToMachinePool(nodeRoles NodeRoles, machinePools []apisV1.RKEM
 		return index, count
 	}
 
+	if nodeRoles.Windows {
+		return len(machinePools) - 1, count
```

Comment:
what if I didn't put windows as the last machinePool though?

Reply:
Is there a reason why you wouldn't, though? I know vSphere allows Windows with Linux out the gate, but with typical K8s clusters, Windows has to come after a cluster is already active with Linux nodes. I would expect we follow the same standard where Windows comes last.

Reply:
rancher / the framework handles adding that last for the user iirc. Generally its at the end of the list, but there's also the case if I have more than 1 machinePool with windows nodes with custom quantities.

Is there a way to look it up instead?

Reply:
Addressed. There is an enhancement that I will make in a separate PR to account for multiple machine pools with the same role. That's a bit outside the scope of this PR.